### DISCUSSION TITLE
How do I use a custom environment by specifying a file path to it

### DISCUSSION THREAD

**Discussion by Eldemarkki** [vitest-dev/vitest#3921] [Q&A] [answered]
In the [Custom Environment documentation](https://vitest.dev/guide/environment.html#custom-environment), it says that I can "_specify a path to a valid JS file (supported since 0.34.0)_". However, I'm not able to get that working. I have the following `vite.config.ts` configuration:

```ts
import { configDefaults, defineConfig } from "vitest/config";

export default defineConfig({
  test: {
    exclude: [...configDefaults.exclude, "data"],
    globals: true,
    environment: "./recipeamigo-test-environment.js",
  },
});
```

When I then try to run `vitest run --threads false`, it says that `Can not find dependency 'vitest-environment-./recipeamigo-test-environment.js` and prompts me to install it as an npm package. I've tried with many different variations, including specifying an absolute path, including the `file://` protocol at the start, just specifying the filename (it's at the same directory level as my `vite.config.ts` and where I run `npm test` from):

![image](https://github.com/vitest-dev/vitest/assets/38920928/6c004fa9-c219-4dae-9420-3290ef9678e6)

I'm using the latest version of Vitest on Linux:
![image](https://github.com/vitest-dev/vitest/assets/38920928/4d4d649b-30c2-40f6-88d5-8c8b2c9ed81b)

Here's the contents of `recipeamigo-test-environment.js` just in case it matters, though it's probably not relevant here:
```js
import { execSync } from "child_process";

const RecipeamigoTestEnvironment = {
  name: "recipeamigo",
  transformMode: "ssr",
  setup: () => {
    execSync("npm run test:start_database");
    return {
      teardown: () => {
        execSync("npm run test:stop_database");
      },
    };
  },
};

export default RecipeamigoTestEnvironment;
```

**Comment by sheremet-va** [comment, marked as answer]
Looks like it's a Vitest bug. But what you are doing in the env doesn't look appropriate for an environment. If you need to run command once, use `globalSetup` instead

**Reply by Eldemarkki** [reply]
Thanks for the quick reply! I didn't know about `globalSetup`, it definitely sounds more appropriate.
